### ISSUE TITLE
[P1][FUND-042] Build Fund Similarity Model

### ISSUE THREAD

**Issue by aaronspindler** [aaronspindler/aaronspindler.com#284]
**Priority**: P1 | **Effort**: XL | **Epic**: Machine Learning & Recommendations
**Dependencies**: FUND-041

## Approach
- Feature engineering from fund characteristics
- Cosine similarity or clustering
- Regular retraining

## Acceptance Criteria
- [ ] Returns relevant similar funds
- [ ] Similarity scores explained
- [ ] Model accuracy > 80%

**Comment by aaronspindler** [comment]
## Implementation Details

### Architecture

```python
# feefifofunds/ml/similarity.py
import numpy as np
from sklearn.preprocessing import StandardScaler
from sklearn.metrics.pairwise import cosine_similarity

class FundSimilarityModel:
    """Find similar funds based on characteristics."""
    
    FEATURES = [
        'expense_ratio',
        'return_1y',
        'return_3y',
        'volatility',
        'sharpe_ratio',
        'beta',
        # Category encoded as one-hot
    ]
    
    def __init__(self):
        self.scaler = StandardScaler()
        self.feature_matrix = None
        self.fund_ids = None
    
    def fit(self, funds_queryset):
        """Build similarity matrix from fund data."""
        features = []
        self.fund_ids = []
        
        for fund in funds_queryset:
            feature_vec = self._extract_features(fund)
            features.append(feature_vec)
            self.fund_ids.append(fund.id)
        
        self.feature_matrix = self.scaler.fit_transform(np.array(features))
        return self
    
    def find_similar(self, fund_id, n=10):
        """Find n most similar funds."""
        idx = self.fund_ids.index(fund_id)
        fund_features = self.feature_matrix[idx].reshape(1, -1)
        
        similarities = cosine_similarity(fund_features, self.feature_matrix)[0]
        
        # Get top n (excluding self)
        similar_indices = np.argsort(similarities)[::-1][1:n+1]
        
        return [
            (self.fund_ids[i], similarities[i])
            for i in similar_indices
        ]
    
    def _extract_features(self, fund):
        # Extract numeric features from fund
        pass
```

### Dependencies
```
scikit-learn>=1.3.0
numpy>=1.24.0
```

### Testing Checklist
- [ ] Returns relevant similar funds
- [ ] Similarity scores explained
- [ ] Model accuracy > 80%
- [ ] Regular retraining scheduled
